internal/services/contracts_events_consumer.go

```diff
@@ -240,3 +247,55 @@ func (c *ContractsEventsConsumer) setMintedAfterMarketDevice(e *ContractEventDat
 
 	return nil
 }
+
+func (c *ContractsEventsConsumer) beneficiarySet(e *ContractEventData) error {
+	var args contracts.RegistryBeneficiarySet
+	err := json.Unmarshal(e.Arguments, &args)
+	if err != nil {
+		return err
+	}
+
+	c.log.Info().Int64("nodeID", args.NodeId.Int64()).Msgf("Aftermarket beneficiary set: %s.", args.Beneficiary)
+
+	device, err := models.AutopiUnits(
+		models.AutopiUnitWhere.AutopiUnitID.EQ(args.NodeId.String()),
+	).One(context.Background(), c.db.DBS().Reader)
+
+	cols := models.AutopiUnitColumns
+
+	device.Beneficiary = null.BytesFrom(args.Beneficiary[:])
```

Comment:
I think we do want to handle the case of `args.Beneficiary == common.Address{}` differently, and in that case set this field to null.